mesos_slave/datadog_checks/mesos_slave/mesos_slave.py

```diff
@@ -124,7 +124,7 @@ def _get_json(self, url, timeout, verify, tags=None):
             status = AgentCheck.CRITICAL
         finally:
             self.log.debug('Request to url : {0}, timeout: {1}, message: {2}'.format(url, timeout, msg))
-            if self.service_check_needed:
+            if self.service_check_needed and not failure_expected:
```

Comment:
Looks like it does not submit an `OK` service check anymore, the test are not passing for the master integration. Probably similar here